bridge/setu/processor/clerk.go

```diff
@@ -95,6 +95,11 @@ func (cp *ClerkProcessor) sendStateSyncedToHeimdall(eventName string, logBytes s
 			"blockNumber", vLog.BlockNumber,
 		)
 
+		if len(event.Data) > helper.MaxStateSyncSize {
+			cp.Logger.Info("Data is too large", "data", hex.EncodeToString(event.Data))
+			event.Data = hmTypes.HexToHexBytes("")
```

Comment:
Why are we resetting the data? Shouldn't we just return an error?

Reply:
we can't return the error
if we return the error then there will be a nonce issue between the network and contracts which halt the state sync process
so we are resetting the data to "" so that we continue the state syncing